src/diffusers/utils/peft_utils.py

```diff
@@ -166,7 +173,8 @@ def get_peft_kwargs(rank_dict, network_alpha_dict, peft_state_dict, is_unet=True
             else:
                 alpha_pattern = {".".join(k.split(".down.")[0].split(".")[:-1]): v for k, v in alpha_pattern.items()}
         else:
-            lora_alpha = set(network_alpha_dict.values()).pop()
+            if not alpha_retrieved:
```

Comment:
Could be merged with previous line: `elif not alpha_retrieved:`.

Note that `lora_alpha` could be undefined if none of the conditions are met, resulting in a `NameError` (this was already true for the previous code). Probably that should never happen, but maybe this could be caught and an appropriate error be raised.

Reply:
> Could be merged with previous line: elif not alpha_retrieved:.

I don't understand this one. How it could be clubbed with the previous line? The main `if/else` conditions are pretty distinct here. 

Regarding the conditions not being met, I agree that this should never happen but I don't have a clear understanding of how to incorporate it without seeing an actual error stemming from it. 

`r = lora_alpha = list(rank_dict.values())[0]` always assigns a `lora_alpha`. So I am not sure how it could go undefined. If we do:

```python
# Try to retrive config.
alpha_retrieved = False
if config is not None:
    lora_alpha = config["lora_alpha"]
    if lora_alpha is None:
        raise ValueError("...")
    alpha_retrieved = True
```

Would it make more sense?

Reply:
Done in 09618d09.